pyxray/parser/deslattes2005.py

```diff
@@ -0,0 +1,204 @@
+    "S": Element(16), "Cl": Element(17), "Ar": Element(18), "K": Element(19), "Ca": Element(20), "Sc": Element(21),
+    "Ti": Element(22), "V": Element(23), "Cr": Element(24), "Mn": Element(25), "Fe": Element(26), "Co": Element(27),
+    "Ni": Element(28), "Cu": Element(29), "Zn": Element(30), "Ga": Element(31), "Ge": Element(32), "As": Element(33),
+    "Se": Element(34), "Br": Element(35), "Kr": Element(36), "Rb": Element(37), "Sr": Element(38), "Y": Element(39),
+    "Zr": Element(40), "Nb": Element(41), "Mo": Element(42), "Tc": Element(43), "Ru": Element(44), "Rh": Element(45),
+    "Pd": Element(46), "Ag": Element(47), "Cd": Element(48), "In": Element(49), "Sn": Element(50), "Sb": Element(51),
+    "Te": Element(52), "I": Element(53), "Xe": Element(54), "Cs": Element(55), "Ba": Element(56), "La": Element(57),
+    "Ce": Element(58), "Pr": Element(59), "Nd": Element(60), "Pm": Element(61), "Sm": Element(62), "Eu": Element(63),
+    "Gd": Element(64), "Tb": Element(65), "Dy": Element(66), "Ho": Element(67), "Er": Element(68), "Tm": Element(69),
+    "Yb": Element(70), "Lu": Element(71), "Hf": Element(72), "Ta": Element(73), "W": Element(74), "Re": Element(75),
+    "Os": Element(76), "Ir": Element(77), "Pt": Element(78), "Au": Element(79), "Hg": Element(80), "Tl": Element(81),
+    "Pb": Element(82), "Bi": Element(83), "Po": Element(84), "At": Element(85), "Rn": Element(86), "Fr": Element(87),
+    "Ra": Element(88), "Ac": Element(89), "Th": Element(90), "Pa": Element(91), "U": Element(92), "Np": Element(93),
+    "Pu": Element(94), "Am": Element(95), "Cm": Element(96), "Bk": Element(97), "Cf": Element(98), "Es": Element(99),
+    "Fm": Element(100),
+}
+
+_SUBSHELL_LOOKUP = {
+    "K": base.K,
+    "L1": base.L1,
+    "L2": base.L2,
+    "L3": base.L3,
+    "M1": base.M1,
+    "M2": base.M2,
+    "M3": base.M3,
+    "M4": base.M4,
+    "M5": base.M5,
+    "N1": base.N1,
+    "N2": base.N2,
+    "N3": base.N3,
+    "N4": base.N4,
+    "N5": base.N5,
+    "N6": base.N6,
+    "N7": base.N7,
+}
+
+_BLEND_LOOKUP = {
+    "KL2,3": base.Ka,
+    "KM": XrayTransition(3, None, None, base.K),
+    "KM2,3": base.Kb1_3,
+    "KM4,5": base.Kb5,
+    "KN2,3": base.Kb2,
+    "KN4,5": base.Kb4,
+    "L2,3M1": base.Ll_n,
+    "L3M4,5": base.La,
+    "L1M2,3": base.Lb3_4,
+    "L1N2,3": base.Lg2_3,
+    "L3N4,5": base.Lb2_15,
+    "L3N6,7": base.Lu,
+    "L2N6,7": XrayTransition(4, 3, None, base.L2),
+    "L1N4,5": XrayTransition(4, 2, None, base.L1),
+    "L1N6,7": XrayTransition(4, 3, None, base.L1),
+}
+
+_NUM_ELEMENTS = 91
+_URL = "https://physics.nist.gov/cgi-bin/XrayTrans/search.pl?download=tab&element=All&lower=10&upper=1000000&units=eV"
+
+# Column numbers in table
+_ELE = 0
+_TRANS = 2
+_THEORY = 3
+_DIRECT = 5
+_COMBINED = 7
+_VAPOR = 9
+_BLEND = 11
+
+
+class Deslattes2005Parser(base._Parser):
+
+    def __iter__(self):
+        r = requests.get(_URL, stream=True)
+
+        r_iter = r.iter_lines()
+        while True:
```

Comment:
I would add a comment that this piece of code find the first row and skip the header

Reply:
I'm not sure about the `while` loop. Why not:
```
for line in r_iter:
    row = line.decode("ascii")
    if row.startswith("Ele."):
         break
```

Reply:
I agree the while loop was unnecessary and have amended this part